### PR TITLE
PHP 7.1 types

### PR DESCRIPTION
**Pull request by thomasvargiu**
Added support for PHP 7.1 types.

- Allowed `void` return type
- Allowed `iterable` type
- Allowed nullable arguments
- Allowed nullable return types

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,10 +25,10 @@ matrix:
       env: PHPDOCUMENTOR_REFLECTION_DOCBLOCK="^2.0"
 
 before_script:
+  - travis_retry composer update --no-interaction
```
Comment (stof) [inline .travis.yml]: this change is wrong. It must be done after changing dependencies (other the change is useless)

File: .travis.yml [hunk 2]
```diff
@@ -11,7 +11,6 @@ cache:
 branches:
   except:
     - /^bugfix\/.*$/
-    - /^feature\/.*$/
```
Comment (stof) [inline .travis.yml]: this must be reverted. We don't want such branches in the main repo to trigger Travis builds for the branch itself (they will be tested by the PR build)

File: composer.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     },
 
     "require-dev": {
-        "phpspec/phpspec": "^2.0",
+        "phpspec/phpspec": "^2.0 || ^3.0",
```
Comment (stof) [inline composer.json]: this is wrong, as the spec suite in the master branch is not compatible with phpspec 3 (there is a pending PR migrating to phpspec 3, but it clearly does not belong in this PR)

**Comment by stof** [comment]
It looks like your PR contains the contributions of @prolic in #287 but under your own authorship. This is not good.

File: composer.json (hunk 3 above)
Comment (thomasvargiu) [inline composer.json]: Without this, tests doesn't work in PHP >= 5.6. Any solutions?

**Comment by thomasvargiu** [comment]
@stof Oh, I didn't see it.  
@prolic, maybe I can integrate my other changes in you PR?

**Comment by prolic** [comment]
Sure. Just submit a pr to my branch.

Am 13.12.2016 19:37 schrieb "Thomas Mauro Vargiu" <notifications@github.com

**Comment by thomasvargiu** [comment]
@stof @prolic. I think my PR is more complete with PHP 7.1 support.
#287 allows only void return type, this PR allows all PHP 7.1 changes: `void` return type, nullable arguments, nullable return types, `iterable` type.

@prolic what do you think?

**Comment by thomasvargiu** [comment]
Ok, I created PR prolic/prophecy#1.

Closed in favor of #287

File: composer.json (hunk 3 above)
Comment (thomasvargiu) [inline composer.json]: @ciaranmcnulty We have the same issue in #287 . You can see on this build: https://travis-ci.org/phpspec/prophecy/builds/183575285

Comment (ciaranmcnulty) [inline composer.json]: While it would make sense to upgrade the suite to work with PhpSpec 3, this is a bug in that test, in prophecy, or an edge case in phpspec that should be fixed. PhpSpec 2 is still in bugfix support for another 6 months

Comment (ciaranmcnulty) [inline composer.json]: this is broken on master - I'll take a look
